ibis/expr/operations/relations.py

```diff
@@ -743,6 +935,19 @@ class ExistsSubquery(Node):
     def output_type(self):
         return ir.ExistsExpr
 
+    def __component_eq__(
+        self,
+        other: ExistsSubquery,
+        cache: MutableMapping[Hashable, bool],
+    ) -> bool:
+        return self.foreign_table.equals(
+            other.foreign_table,
+            cache=cache,
+        ) and all(
+            left_pred.equals(right_pred, cache=cache)
```

Comment:
maybe with a helper function to avoid writing all of the left right equals repeatedly

Reply:
Done. Added `util.seq_eq` for comparing sequences of `CachedEqMixin` subclasses.